src/FilterBundle/Normalizer/Term/EnumAttributeNormalizer.php

```diff
@@ -136,22 +169,12 @@ private function getProductTypes(): array
         if (!is_array($this->productTypes)) {
             $this->productTypes = [];
 
-            do {
-                $productTypeRequest = ProductTypeQueryRequest::of();
-                $productTypeRequest->limit(500);
-
-                $response = $this->client->execute($productTypeRequest);
-
-                if ($response instanceof PagedQueryResponse) {
-                    $productTypeCollection = $response->toObject();
+            $request = ProductTypeQueryRequest::of();
 
-                    if ($productTypeCollection instanceof ProductTypeCollection) {
-                        foreach ($productTypeCollection as $productType) {
-                            $this->productTypes[] = $productType;
-                        }
-                    }
-                }
-            } while ($response->getTotal() > ($response->getOffset() + $response->getCount()));
+            /** @var ProductTypeCollection $productType */
```

Comment:
Wie besprochen kann ich hier noch was gemacht werden ;)

Reply:
Angepasst ... Code ist jetzt auf 

`$this->productTypes = iterator_to_array($this->queryHelper->getAll($this->client, $request));``

geschrumpft, womit auch der DocBlock seine Daseinsberechtigung verliert.